bin/test/i18n/resources/dashboard/course_content/test_sync_in.rb

```diff
@@ -0,0 +1,711 @@
+    exec_seq = sequence('execution')
+
+    Dir.expects(:[]).with(CDO.dir('i18n/locales/source/course_content/**/*.json')).in_sequence(exec_seq).returns([CDO.dir('i18n/locales/source/course_content/blank_i18n_source_file.json'), CDO.dir('i18n/locales/source/course_content/expected_i18n_source_file.json')])
+
+    # 'i18n/locales/source/course_content/blank_i18n_source_file.json' should be skipped
+    JSON.expects(:load_file).with(CDO.dir('i18n/locales/source/course_content/blank_i18n_source_file.json')).in_sequence(exec_seq).returns({})
+    sync_in_instance.expects(:select_redactable).with({}).never
+
+    JSON.expects(:load_file).with(CDO.dir('i18n/locales/source/course_content/expected_i18n_source_file.json')).in_sequence(exec_seq).returns({'expected_level_url' => 'expected_i18n_strings'})
+    sync_in_instance.expects(:select_redactable).with('expected_i18n_strings').in_sequence(exec_seq).returns({'expected_redactable_data' => 'expected_redactable_i18n_strings'})
+
+    FileUtils.expects(:mkdir_p).with(CDO.dir('i18n/locales/original/course_content')).in_sequence(exec_seq)
+    File.expects(:write).with(CDO.dir('i18n/locales/original/course_content/expected_i18n_source_file.json'), %Q[{\n  "expected_level_url": {\n    "expected_redactable_data": "expected_redactable_i18n_strings"\n  }\n}]).in_sequence(exec_seq)
+
+    RedactRestoreUtils.expects(:redact_data).with({'expected_level_url' => {'expected_redactable_data' => 'expected_redactable_i18n_strings'}}, %w[blockly]).in_sequence(exec_seq).returns({'expected_level_url' => {'expected_redactable_data' => 'expected_redacted_i18n_strings'}})
+    File.expects(:write).with(CDO.dir('i18n/locales/source/course_content/expected_i18n_source_file.json'), %Q[{\n  "expected_level_url": {\n    "expected_redactable_data": "expected_redacted_i18n_strings"\n  }\n}]).in_sequence(exec_seq)
+
+    sync_in_instance.send(:redact_level_content)
+  end
+
+  # def test_dsl_level_preparation
+  #   ActiveRecord::Base.transaction do
+  #     Unit.destroy_all
+  #     Level.destroy_all
+  #
+  #     script = FactoryBot.create(:unit, name: 'coursea-2017')
+  #     lesson_group = FactoryBot.create(:lesson_group, script: script)
+  #     lesson = FactoryBot.create(:lesson, lesson_group: lesson_group)
+  #
+  #     child_level = FactoryBot.create(
+  #       :free_response,
+  #       name: 'Child DSLDefined level',
+  #       display_name: 'Child DSLDefined level display_name',
+  #       bubble_choice_description: 'Child DSLDefined level bubble_choice_description',
+  #       short_instructions: 'Child DSLDefined level short_instructions',
+  #       long_instructions: 'Child DSLDefined level long_instructions',
+  #       # failure_message_override: 'Child DSLDefined level failure_message_overrides',
+  #       teacher_markdown: 'Child DSLDefined level teacher_markdown',
+  #       placeholder: 'Child DSLDefined level placeholder',
+  #       title: 'Child DSLDefined level title',
+  #     )
```

Comment:
Double checking that this is needed here

Reply:
No, sorry. Will delete this